opensrp-reveal/src/main/java/org/smartregister/reveal/util/InteractorUtils.java

```diff
@@ -110,4 +122,87 @@ public boolean archiveClient(String baseEntityId, boolean isFamily) {
         }
         return saved;
     }
+
+    public boolean archiveEventsForTask(SQLiteDatabase db, BaseTaskDetails taskDetails) {
+        boolean archived = true;
+        Task task = taskRepository.getTaskByIdentifier(taskDetails.getTaskId());
+
+        if (task == null) {
+            return false;
+        }
+        List<String> formSubmissionIds = getFormSubmissionIdsFromEventTask(db, task.getIdentifier());
+        List<EventClient> eventClients = eventClientRepository.fetchEventClients(formSubmissionIds);
+        DateTime now = new DateTime();
+        JSONArray taskEvents = new JSONArray();
+
+        if(eventClients == null || eventClients.isEmpty()) {
+            return false;
+        }
+
+        try {
+
+            for (EventClient eventClient: eventClients) {
+                org.smartregister.domain.db.Event event = eventClient.getEvent();
+                JSONObject eventJson = new JSONObject(gson.toJson(event));
+                eventJson.put("dateVoided", now);
+                eventJson.put(EventClientRepository.event_column.syncStatus.name(), BaseRepository.TYPE_Unsynced);
+                taskEvents.put(eventJson);
+            }
+
+            eventClientRepository.batchInsertEvents(taskEvents, 0);
+
+            Event resetTaskEvent = RevealJsonFormUtils.createTaskEvent(task.getForEntity(), Utils.getCurrentLocationId(),
+                    null, Constants.TASK_RESET_EVENT, Constants.STRUCTURE);
+            JSONObject eventJson = new JSONObject(gson.toJson(resetTaskEvent));
+            eventJson.put(EventClientRepository.event_column.syncStatus.name(), BaseRepository.TYPE_Unsynced);
+
+            populateEventDetails(eventJson, task.getIdentifier(), taskDetails.getBusinessStatus(), taskDetails.getTaskStatus(), taskDetails.getStructureId());
+            eventClientRepository.addEvent(task.getForEntity(), eventJson);
+
+
+        } catch (Exception e) {
+            Timber.e(e);
+            archived = false;
+        }
+
+        return archived;
+    }
+
+    List<String> getFormSubmissionIdsFromEventTask(SQLiteDatabase db, String taskIdentifier) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://app.codacy.com/manual/OpenSRP/opensrp-client-reveal/pullRequest?prid=4997357)